2025-03-22 22:02:01,612 - INFO - Starting bot and syncing channel members...
2025-03-22 22:02:01,623 - INFO - Starting callback server on 127.252.4.177:8001
2025-03-22 22:02:01,761 - INFO - Found 2 administrators
2025-03-22 22:02:01,762 - INFO - Channel members synced with database
2025-03-22 22:02:01,762 - INFO - Bot initialization complete.
2025-03-22 22:02:01,804 - INFO - Webhook removed successfully
2025-03-22 22:02:01,805 - INFO - Starting polling...
2025-03-22 22:02:33,187 - INFO - Received message from user 79100530
2025-03-22 22:02:33,602 - INFO - User 79100530 status: creator
2025-03-22 22:02:33,835 - INFO - User 79100530 is a member. Sent membership info.
2025-03-22 22:03:25,932 - INFO - Received message from user 6513509854
2025-03-22 22:03:26,055 - INFO - User 6513509854 status: left
2025-03-22 22:03:26,055 - INFO - User 6513509854 is not an active member (status: left). Checking payment status.
2025-03-22 22:03:26,890 - INFO - ZarinPal response for user 6513509854: {'data': {'authority': 'A0000000000000000000000000003z5gg7pe', 'fee': 3000, 'fee_type': 'Merchant', 'code': 100, 'message': 'Success'}, 'errors': []}
2025-03-22 22:03:27,031 - INFO - Sent payment link to user 6513509854: https://www.zarinpal.com/pg/StartPay/A0000000000000000000000000003z5gg7pe
2025-03-22 22:04:10,261 - INFO - Received message from user 6513509854
2025-03-22 22:04:10,788 - INFO - User 6513509854 status: left
2025-03-22 22:04:10,788 - INFO - User 6513509854 is not an active member (status: left). Checking payment status.
2025-03-22 22:04:11,326 - INFO - ZarinPal response for user 6513509854: {'data': {'authority': 'A0000000000000000000000000007z5vvmqw', 'fee': 3000, 'fee_type': 'Merchant', 'code': 100, 'message': 'Success'}, 'errors': []}
2025-03-22 22:04:11,578 - INFO - Sent payment link to user 6513509854: https://www.zarinpal.com/pg/StartPay/A0000000000000000000000000007z5vvmqw
2025-03-22 23:22:50,869 - INFO - Received message from user 79100530
2025-03-22 23:22:51,042 - INFO - User 79100530 status: creator
2025-03-22 23:22:51,172 - INFO - User 79100530 is a member. Sent membership info.
2025-03-23 00:34:26,355 - INFO - Received message from user 7425853813
2025-03-23 00:34:26,513 - INFO - User 7425853813 status: left
2025-03-23 00:34:26,514 - INFO - User 7425853813 is not an active member (status: left). Checking payment status.
2025-03-23 00:34:27,348 - INFO - ZarinPal response for user 7425853813: {'data': {'authority': 'A000000000000000000000000000geyxqo3g', 'fee': 3000, 'fee_type': 'Merchant', 'code': 100, 'message': 'Success'}, 'errors': []}
2025-03-23 00:34:27,463 - INFO - Sent payment link to user 7425853813: https://www.zarinpal.com/pg/StartPay/A000000000000000000000000000geyxqo3g
2025-03-23 00:34:38,638 - INFO - Received message from user 6513509854
2025-03-23 00:34:38,796 - INFO - User 6513509854 status: left
2025-03-23 00:34:38,796 - INFO - User 6513509854 is not an active member (status: left). Checking payment status.
2025-03-23 00:34:39,523 - INFO - ZarinPal response for user 6513509854: {'data': {'authority': 'A000000000000000000000000000nglm3ep3', 'fee': 3000, 'fee_type': 'Merchant', 'code': 100, 'message': 'Success'}, 'errors': []}
2025-03-23 00:34:39,673 - INFO - Sent payment link to user 6513509854: https://www.zarinpal.com/pg/StartPay/A000000000000000000000000000nglm3ep3
2025-03-23 00:35:11,122 - INFO - Received message from user 6513509854
2025-03-23 00:35:11,159 - INFO - User 6513509854 status: left
2025-03-23 00:35:11,159 - INFO - User 6513509854 is not an active member (status: left). Checking payment status.
2025-03-23 00:35:11,756 - INFO - ZarinPal response for user 6513509854: {'data': {'authority': 'A0000000000000000000000000002q5r78zl', 'fee': 3000, 'fee_type': 'Merchant', 'code': 100, 'message': 'Success'}, 'errors': []}
2025-03-23 00:35:11,860 - INFO - Sent payment link to user 6513509854: https://www.zarinpal.com/pg/StartPay/A0000000000000000000000000002q5r78zl
2025-03-23 00:35:33,364 - INFO - Received message from user 7425853813
2025-03-23 00:35:33,404 - INFO - User 7425853813 status: left
2025-03-23 00:35:33,404 - INFO - User 7425853813 is not an active member (status: left). Checking payment status.
2025-03-23 00:35:33,964 - INFO - ZarinPal response for user 7425853813: {'data': {'authority': 'A000000000000000000000000000rv6wqp6z', 'fee': 3000, 'fee_type': 'Merchant', 'code': 100, 'message': 'Success'}, 'errors': []}
2025-03-23 00:35:34,083 - INFO - Sent payment link to user 7425853813: https://www.zarinpal.com/pg/StartPay/A000000000000000000000000000rv6wqp6z
2025-03-23 00:36:05,780 - INFO - Received message from user 1949259578
2025-03-23 00:36:05,837 - INFO - User 1949259578 status: left
2025-03-23 00:36:05,837 - INFO - User 1949259578 is not an active member (status: left). Checking payment status.
2025-03-23 00:36:06,377 - INFO - ZarinPal response for user 1949259578: {'data': {'authority': 'A000000000000000000000000000eexr6q81', 'fee': 3000, 'fee_type': 'Merchant', 'code': 100, 'message': 'Success'}, 'errors': []}
2025-03-23 00:36:06,525 - INFO - Sent payment link to user 1949259578: https://www.zarinpal.com/pg/StartPay/A000000000000000000000000000eexr6q81
2025-03-23 10:51:26,776 - INFO - Received message from user 79100530
2025-03-23 10:51:26,909 - INFO - User 79100530 status: creator
2025-03-23 10:51:26,977 - INFO - User 79100530 is a member. Sent membership info.
Traceback (most recent call last):
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 534, in _make_request
    response = conn.getresponse()
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/urllib3/connection.py", line 516, in getresponse
    httplib_response = super().getresponse()
  File "/usr/lib/python3.10/http/client.py", line 1375, in getresponse
    response.begin()
  File "/usr/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.10/http/client.py", line 279, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib/python3.10/socket.py", line 705, in readinto
    return self._sock.recv_into(b)
  File "/usr/lib/python3.10/ssl.py", line 1303, in recv_into
    return self.read(nbytes, buffer)
  File "/usr/lib/python3.10/ssl.py", line 1159, in read
    return self._sslobj.read(len, buffer)
TimeoutError: The read operation timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 841, in urlopen
    retries = retries.increment(
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/urllib3/util/retry.py", line 474, in increment
    raise reraise(type(error), error, _stacktrace)
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/urllib3/util/util.py", line 39, in reraise
    raise value
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 787, in urlopen
    response = self._make_request(
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 536, in _make_request
    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 367, in _raise_timeout
    raise ReadTimeoutError(
urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='api.telegram.org', port=443): Read timed out. (read timeout=25)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/ICAM_Bot/G3_bot.py", line 316, in <module>
    remove_webhook_and_start()
  File "/root/ICAM_Bot/G3_bot.py", line 306, in remove_webhook_and_start
    bot.polling(none_stop=True)
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/telebot/__init__.py", line 1198, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/telebot/__init__.py", line 1273, in __threaded_polling
    raise e
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/telebot/__init__.py", line 1234, in __threaded_polling
    polling_thread.raise_exceptions()
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/telebot/util.py", line 111, in raise_exceptions
    raise self.exception_info
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/telebot/util.py", line 93, in run
    task(*args, **kwargs)
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/telebot/__init__.py", line 688, in __retrieve_updates
    updates = self.get_updates(offset=(self.last_update_id + 1), 
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/telebot/__init__.py", line 660, in get_updates
    json_updates = apihelper.get_updates(
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/telebot/apihelper.py", line 332, in get_updates
    return _make_request(token, method_url, params=payload)
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/telebot/apihelper.py", line 162, in _make_request
    result = _get_req_session().request(
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/root/ICAM_Bot/G3_bot_venv/lib/python3.10/site-packages/requests/adapters.py", line 713, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='api.telegram.org', port=443): Read timed out. (read timeout=25)
